server/app/scanpy_engine/scanpy_engine.py

```diff
@@ -480,7 +411,9 @@ def data_frame_to_fbs_matrix(self, filter, axis):
             raise FilterError("filtering on obs unsupported")
 
         # Currently only handles VAR dimension
-        X = self.data._X[:, var_selector]
+        X = self.data._X
+        if var_selector is not None:
+            X = X[:, var_selector]
```

Comment:
Is this related to the nan-to-num or is this just an improvement?

Reply:
latent bug required to support the test cases (not used by the front-end, but should be supported).     This was noted in the PR comments:  "fixed bug in dataframe filtering for FBS route (did not correctly handle a filter of None)"

Reply:
Ahh, I understand now.